docs/DeveloperGuide.md

```diff
@@ -1,38 +1,726 @@
+        * [Target user profile](#target-user-profile)
+        * [Value proposition](#value-proposition)
+    * [User Stories](#user-stories)
+    * [Non-Functional Requirements](#non-functional-requirements)
+    * [Glossary](#glossary)
+    * [Instructions for manual testing](#instructions-for-manual-testing)
+        * [Launch](#launch)
+        * [Sample test cases](#sample-test-cases)
+            * [Help command](#help-command)
+            * [Get reflection questions](#get-reflection-questions)
+            * [Add atomic habits](#add-atomic-habits)
+        * [Saving data](#saving-data)
+
+<!-- TOC -->
+
+## Product Name
+
+**WellNUS++**
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+1. Reference to AB-3 Developer Guide: https://se-education.org/addressbook-level3/DeveloperGuide.html
+2. Reference to AB-3 diagrams code: https://github.com/se-edu/addressbook-level3/tree/master/docs/diagrams
+
+## Setting up, getting started
+
+### Setting up the project in your computer
+
+Firstly, **fork** this repo, and **clone** the fork into your computer. <br>
+<br>
+If you plan to use Intellij IDEA (highly recommended): <br>
+
+1. **Configure the JDK**: Follow the guide
+   [[se-edu/guides] IDEA: Configuring the JDK](https://se-education.org/guides/tutorials/intellijJdk.html)
+   to ensure Intellij is configured to use **JDK 11**.
+2. **Import the project as a Gradle project**: Follow the guide
+   [[se-edu/guides] IDEA: Importing a Gradle project](https://se-education.org/guides/tutorials/intellijImportGradleProject.html)
+   to import the project into IDEA.<br>
+   **Note**: Importing a Gradle project is slightly different from importing a normal Java project.
+3. **Verify the setup:**
+    1. Run the ```wellnus.WellNus``` and try a few commands.
+    2. Run the tests to ensure they all pass.
+
+### Before writing code
+
+1. **Configure the coding style**<br>
+   If using IDEA, follow the guide
+   [[se-edu/guides] IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html)
+   to set up IDEA’s coding style to match ours.<br>
+   <br>
+2. **Set up CI**<br>
+   This project comes with a GitHub Actions config files (in `.github/workflows` folder).
+   When GitHub detects those files, it will run the CI for your project automatically at each push
+   to the `master` branch or to any PR. No set up required.<br>
+   <br>
+3. **Learn the design**<br>
+   When you are ready to start coding, we recommend that you look at the class diagrams to understand the structure of
+   the
+   code and the interaction among different classes.<br>
 
 ## Design & implementation
+### Application Lifecycle
+#### Overview
+The overall execution lifecycle of the WellNus application involves 4 main components, as shown in the diagram below.
+
+![Application Lifecycle](diagrams/WellnusSequence.png)
```

Comment:
![image](https://user-images.githubusercontent.com/88386677/228428450-1445a4de-74da-4c52-b35c-76ceed446ea0.png)
Text on dotted lines should be the variable returned rather than a description?